### PR TITLE
refactor: added additional text and code examples to promise.md file.

### PR DESCRIPTION
**Pull request by Dusch4593**
### Description

Edited `promise.md` file by adding the following sections: 

* Promise object states
* Syntax
* Chaining

Code snippets were also provided because they're nice to have as companion-media for the entries.

### Type of Change

This was a light refactor where nothing was deleted and only new content was added.

- [ ] Adding a new entry
- [X] Editing an existing entry (fixing a typo, bug, issues, etc)
- [ ] Updating the documentation

### Checklist

- [X] My entry follows the Codecademy Docs style guide
- [X] I have performed a self-review of my own writing and code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have checked my entry and corrected any misspellings
- [X] All writings are my own

### Codecademy Username

@BrandonDusch 
https://github.com/Dusch4593

### CODE REVIEW

File: content/javascript/concepts/promise/promise.md [hunk 1]
```diff
@@ -13,4 +13,65 @@ CatalogContent:
...
+
+Promises are ideal for performing asynchronous JavaScript operations. This is supported by their use of three states: 
+
+* A `pending` state, where the promises has yet to resolve or be rejected.
+* A `fulfilled` status, where asynchronous operations are complete and a value has been returned.
+* A `rejected` state, where asynchronous operations have failed at some point.
+
+### Syntax
+
+A `Promise` object is returned from a function that accepts two unique functins: `resolve` and `reject`:
```
Comment (lukeecart) [inline content/javascript/concepts/promise/promise.md:28]:
```suggestion
A `Promise` object is returned from a function that accepts two unique functions: `resolve` and `reject`: 
````

File: content/javascript/concepts/promise/promise.md [hunk 2]
```diff
@@ -13,4 +13,65 @@ CatalogContent:
...
 A JavaScript `Promise` is an object that can be used to get the outcome of an asynchronous operation when that result is not instantly available.
 
-Since JavaScript code runs in a non-blocking manner, promises become essential when we have to wait for some asynchronous operation without holding back the execution of the rest of the code.
+Since JavaScript code runs in a non-blocking manner, promises become essential when we have to wait for some asynchronous operation without holding back the execution of the rest of the code. Some common use cases for promises include using data from APIs and downloading images.
+
+### Promise object state
+
+Promises are ideal for performing asynchronous JavaScript operations. This is supported by their use of three states: 
+
+* A `pending` state, where the promises has yet to resolve or be rejected.
```
Comment (lukeecart) [inline content/javascript/concepts/promise/promise.md:22]:
```suggestion
* A `pending` state, where the promise has yet to resolve or be rejected.
```

File: content/javascript/concepts/promise/promise.md [hunk 3]
```diff
@@ -13,4 +13,65 @@ CatalogContent:
...
+
+#### Chaining
+
+Chaining is a technique used to perform additional operations against a "fulfilled" promise. The following methods are used for promise chaining: 
+
+* `.then()`, which performs operations on a fulfilled promise object and returns a new value.
+* `.catch()`, which handles any anticipated promise rejections and throws an error.
+* `.finally()`, which executes without regard to whether the promise was fulfilled or rejected.
+
+Below is a brief example of using chaining to produce a sucess message after the promise is fulfilled:
```
Comment (lukeecart) [inline content/javascript/concepts/promise/promise.md:46]:
```suggestion
Below is a brief example of using chaining to produce a success message after the promise is fulfilled:
```

File: content/javascript/concepts/promise/promise.md [hunk 4]
```diff
@@ -13,4 +13,65 @@ CatalogContent:
...
+
+```js
+const promise = new Promise((resolve, reject) => {
+    const term1 = 2;
+    const term2 = 2;
+    let expression = term1 + term2;
+    let answer = 4;
+    let expressionIsCorrect = expression === answer;
```
Comment (lukeecart) [inline content/javascript/concepts/promise/promise.md:54]: Could we change these lets to consts? As they don't seem to change

File: content/javascript/concepts/promise/promise.md [hunk 5]
```diff
@@ -13,4 +13,65 @@ CatalogContent:
...
+
+promise.then(message => {
+    console.log(`Success: ${message}`)
+}).catch(err => {
+    console.log(`${err.errorType}: ${err.message} \nOriginal Expression: ${err.originalExpression}`)
+}).finally(message => {
+    console.log("Operations finished")
+});
+
+// Output: Success: The promise was fulfilled!
```
Comment (lukeecart) [inline content/javascript/concepts/promise/promise.md:76]: Very clear example 👏

**Review by lukeecart — CHANGES_REQUESTED** [review]
Just a few spelling errors but other than that looks good 👌

File: content/javascript/concepts/promise/promise.md (hunk 4 above)
Comment (Dusch4593) [reply content/javascript/concepts/promise/promise.md:54]: Sure!

**Comment by Dusch4593** [comment]
Changes pushed! How do we look now, @lukeecart?

**Review by lukeecart — APPROVED** [review]

**Comment by lukeecart** [comment]
Looks like we need to run 'npm run format:all' again. 
The formating seems to be catching everyone out

**Comment by Dusch4593** [comment]
Hmm this is what I'm getting when I run `npm run format:all` on the `edit-promise-entry` branch: 

```
> prettier --ignore-path .prettierignore --write "./**/*.(ts|md)"

[error] Invalid configuration file `CODE_OF_CONDUCT.md`: Cannot find module '@codecademy/prettier-config'
[error] Require stack:
[error] - /Users/brandondusch/Desktop/work/codecademy/docs/.prettierrc.js
[error] - /usr/local/lib/node_modules/prettier/third-party.js
[error] - /usr/local/lib/node_modules/prettier/index.js
[error] - /usr/local/lib/node_modules/prettier/bin-prettier.js
```

Is there something going on with the current version of `CODE_OF_CONDUCT.md` or something with regards to prettier?

**Comment by lukeecart** [comment]
@Dusch4593 When running locally, it's because you need to run `npm install` to download the prettier config which is an npm module in the package.json file

**Comment by Dusch4593** [comment]
> @Dusch4593 When running locally, it's because you need to run `npm install` to download the prettier config which is an npm module in the package.json file

Yup just realized lol. Everything ran fine :D

**Review by yangc95 — APPROVED** [review]

**Pull request merged**